src/wallet/wallet.cpp

```diff
@@ -417,6 +417,12 @@ bool CWallet::ChangeWalletPassphrase(const SecureString& strOldWalletPassphrase,
 
 void CWallet::SetBestChain(const CBlockLocator& loc)
 {
+    LOCK(cs_wallet); //nWalletMaxVersion
+    unsigned int keypool_min = GetArg("-keypoolmin", DEFAULT_KEYPOOL_MIN);
+    if (IsHDEnabled() && ((CanSupportFeature(FEATURE_HD_SPLIT) && setInternalKeyPool.size() < keypool_min) || (setExternalKeyPool.size() < keypool_min))) {
```

Comment:
Ah, your allusion was a bit subtle for me. I thought your original comment was "this is an alternative way to do it" rather than "the current way is unsafe. You should do this instead"

Can you take a look at https://github.com/jnewbery/bitcoin/commit/145a06521bd458e214cd9c8d163c9a1ebbd63189 . Is that what you had in mind?

With that change, I think we can remove `bypasskeypoolcritical` and update the recovery instructions to say `set -keypool_critical=0`. Do you agree?